finalitygadget/finalitygadget.go

```diff
@@ -83,10 +83,14 @@ func NewFinalityGadget(cfg *config.Config, db db.IDatabaseHandler, logger *zap.L
 		return nil, err
 	}
 
-	lastProcessedHeight, err := initializeLastProcessedHeight(l2Client, db)
+	lastProcessedHeight := uint64(0)
```

Comment:
what if instead of starting from block 0 by default, we query the latest safe block number?

this way we don't have to rely on any trusted peers, while still starting from reasonably close to chain tip

@lesterli @bap2pecs

Reply:
The `lastProcessedHeight` presents the finality gadget processed block height, when the finality gadget has not processed any block yet, I prefer to set it with block 0.

Reply:
discussed w @lesterli offline, it won't always scan from 0 (there are 2 cases)

- when init, it's set to 0
- when first entering the loop case <-ticker.C, if it's not btc activated, it will immediately set it to latest finalied block (e.g. can be very large if let's say we integreate to an exsiting chain)
-  when first entering the loop case <-ticker.C, if it's alredy btc activated, that will have a small problem b/c it will start to handleBlock() from 0. this can be optimized by setting lastProcessedHeight to latestFinalizedHeight-1 at startup. this can avoid indexing from blocks that's not activated yet cc @lesterli to create a task

Reply:
we just need to add

```
if fg.lastProcessedHeight == 0 {
	fg.lastProcessedHeight = latestFinalizedHeight - 1
}
```

ideally it should be

```
if fgconf.syncFull === false && fg.lastProcessedHeight == 0 {
	fg.lastProcessedHeight = latestFinalizedHeight - 1
} else {
	fg.lastProcessedHeight = convert(btcStakingActivatedTimestamp)
}
```
but this can be done later. we can just add a TODO here